aea/decision_maker/messages/transaction.py

```diff
@@ -221,27 +216,77 @@ def check_consistency(self) -> bool:
         return True
 
     @classmethod
-    def respond_with(cls, other: 'TransactionMessage', performative: Performative, transaction_digest: Optional[str] = None) -> 'TransactionMessage':
+    def respond_settlement(cls, other: 'TransactionMessage', performative: Performative, tx_digest: Optional[str] = None) -> 'TransactionMessage':
+        """
+        Create response message.
+
+        :param other: TransactionMessage
+        :param performative: the performative
+        :param tx_digest: the transaction digest
+        :return: a transaction message object
+        """
+        if tx_digest is None:
+            tx_msg = TransactionMessage(performative=performative,
+                                        skill_callback_ids=other.skill_callback_ids,
+                                        tx_id=other.tx_id,
+                                        tx_sender_addr=other.tx_sender_addr,
+                                        tx_counterparty_addr=other.tx_counterparty_addr,
+                                        tx_amount_by_currency_id=other.tx_amount_by_currency_id,
+                                        tx_sender_fee=other.tx_sender_fee,
+                                        tx_counterparty_fee=other.tx_counterparty_fee,
+                                        tx_quantities_by_good_id=other.tx_quantities_by_good_id,
+                                        ledger_id=other.ledger_id,
+                                        info=other.info)
+        else:
+            tx_msg = TransactionMessage(performative=performative,
+                                        skill_callback_ids=other.skill_callback_ids,
+                                        tx_id=other.tx_id,
+                                        tx_sender_addr=other.tx_sender_addr,
+                                        tx_counterparty_addr=other.tx_counterparty_addr,
+                                        tx_amount_by_currency_id=other.tx_amount_by_currency_id,
+                                        tx_sender_fee=other.tx_sender_fee,
+                                        tx_counterparty_fee=other.tx_counterparty_fee,
+                                        tx_quantities_by_good_id=other.tx_quantities_by_good_id,
+                                        ledger_id=other.ledger_id,
+                                        info=other.info,
+                                        tx_digest=tx_digest)
+        return tx_msg
```

Comment:
couldn't that simplified with:
```python
tx_msg = TransactionMessage(...)
if tx_digest is None:
    tx_msg.set("tx_digest", tx_digest)
return tx_msg
```

Reply:
Well it could, but then the consistency check is not applied as this is only run upon the init. We can clean this up in the future to have consistency run also on setters!